codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/codegen/knowledge/GoPointableIndex.java

```diff
@@ -99,8 +96,13 @@ public class GoPointableIndex implements KnowledgeIndex {
             ShapeType.BIG_INTEGER
     );
 
-    // All types types that are comparable to nil
-    private static final Set<ShapeType> NUMBER_SHAPE = SetUtils.of(
+    // Pointer types that are strictly pointer type even if it has required trait
```

Comment:
This is NOT True. If there's a code that assumes this then we need to fix this.

Reply:
Removed STRING shape to be pointer always